### PR TITLE
fix: error in the log when no pd member fails

### PR DESCRIPTION
**Pull request by hoyhbx**
### What problem does this PR solve?

When failovering a broken PD member, [this line](https://github.com/pingcap/tidb-operator/blob/d5ca267ad940d7037954d2185cbe60a023954f3b/pkg/manager/member/pd_failover.go#L70) gives us an error even if no PD member fails (i.e. `pdDeletedFailureReplicas=0`). When we inspect the error log, we find that the error message “**PD failover replicas (0) reaches the limit (0), skip failover**” is constantly written to the operator log and is very confusing to users. However, there is actually no error if no PD member fails. We believe when `pdDeletedFailureReplicas=0`, the operator should not throw the error message “**PD failover replicas (0) reaches the limit (0), skip failover**“.
This PR provides a straightforward fix for this problem.
### What is changed and how does it work?

We fix it by adding one more condition in the if statement in [this line](https://github.com/pingcap/tidb-operator/blob/d5ca267ad940d7037954d2185cbe60a023954f3b/pkg/manager/member/pd_failover.go#L70) to check whether `pdDeletedFailureReplicas` is larger than 0, as illustrated below:
```golang
if pdDeletedFailureReplicas >= *tc.Spec.PD.MaxFailoverCount && pdDeletedFailureReplicas > 0 {
	klog.Errorf(“PD failover replicas (%d) reaches the limit (%d), skip failover”, pdDeletedFailureReplicas, *tc.Spec.PD.MaxFailoverCount)
	return nil
}
```
### Code changes
- [x] Has Go code change
- [ ] Has CI related scripts change
### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 
This is a simple fix and we suppose no test above is needed.
### Side effects
- [ ] Breaking backward compatibility 
- [ ] Other side effects: 
### Related changes
- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 
### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.
```release-note
NONE
```

### CODE REVIEW

File: pkg/manager/member/pd_failover.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (f *pdFailover) Failover(tc *v1alpha1.TidbCluster) error {
 	}
 
 	pdDeletedFailureReplicas := tc.GetPDDeletedFailureReplicas()
-	if pdDeletedFailureReplicas >= *tc.Spec.PD.MaxFailoverCount {
+	if pdDeletedFailureReplicas >= *tc.Spec.PD.MaxFailoverCount && pdDeletedFailureReplicas > 0 {
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_failover.go:69]: For your case, you have set `Spec.PD.MaxFailoverCount=0`, right? Which means that you want to disable failover for PD, in this case, I think we should update the code here https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_member_manager.go#L241 as below:
`if m.deps.CLIConfig.AutoFailover && tc.Spec.PD.MaxFailoverCount != nil && *tc.Spec.PD.MaxFailoverCount > 0 {`
WDYT?

Comment (hoyhbx) [reply pkg/manager/member/pd_failover.go]: @DanielZhangQD Thanks for your reply. Yes, we think that your suggestion is correct, and we have updated the code in the latest commit in this PR now.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by KanShiori** [comment]
What happens if failover occurs and set the max failover count to 0?

File: pkg/manager/member/pd_failover.go (hunk 1 above)
Comment (KanShiori) [reply pkg/manager/member/pd_failover.go]: What happens if failover occurs and set the max failover count to 0?

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func (m *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClust
 		return err
 	}
 
-	if m.deps.CLIConfig.AutoFailover {
+	if m.deps.CLIConfig.AutoFailover && tc.Spec.PD.MaxFailoverCount != nil && *tc.Spec.PD.MaxFailoverCount > 0 {
```
Comment (KanShiori) [inline pkg/manager/member/pd_member_manager.go:241]: What happens if failover occurs and set the max failover count to 0?

Comment (KanShiori) [reply pkg/manager/member/pd_member_manager.go:241]: can not arrive the `Recover` func?

Comment (DanielZhangQD) [reply pkg/manager/member/pd_member_manager.go:241]: The failover Pods will not be scaled in and the user needs to remove the failure member manually in the status.pd then the failover Pods will be scaled in.

Comment (DanielZhangQD) [reply pkg/manager/member/pd_member_manager.go:241]: @hoyhbx After checking the logic of the other component, I think @KanShiori 's above comment is reasonable, we should be able to recover the previous failover Pods even if we set MaxFailoverCount to 0.

Comment (DanielZhangQD) [reply pkg/manager/member/pd_member_manager.go:241]: Could you please add the check in L244?

Comment (hoyhbx) [reply pkg/manager/member/pd_member_manager.go]: Sure, I have changed it. @DanielZhangQD

File: pkg/manager/member/pd_member_manager.go [hunk 3]
```diff
@@ -241,7 +241,7 @@ func (m *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClust
 	if m.deps.CLIConfig.AutoFailover {
 		if m.shouldRecover(tc) {
 			m.failover.Recover(tc)
-		} else if tc.PDAllPodsStarted() && !tc.PDAllMembersReady() || tc.PDAutoFailovering() {
+		} else if (tc.PDAllPodsStarted() && !tc.PDAllMembersReady() || tc.PDAutoFailovering()) && tc.Spec.PD.MaxFailoverCount != nil && *tc.Spec.PD.MaxFailoverCount > 0 {
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_member_manager.go:244]:
```suggestion
		} else if tc.Spec.PD.MaxFailoverCount != nil && *tc.Spec.PD.MaxFailoverCount > 0 && (tc.PDAllPodsStarted() && !tc.PDAllMembersReady() || tc.PDAutoFailovering()) {
```

**Review by KanShiori — APPROVED** [review]

**Pull request merged**